cxx-checks/src/test/java/org/sonar/cxx/checks/LineRegularExpressionCheckTest.java

```diff
@@ -20,77 +20,123 @@
+    String content = new String(Files.readAllBytes(new File(sensorContext.fileSystem().baseDir(), fileName).toPath()), "UTF-8");
+    sensorContext.fileSystem().add(new DefaultInputFile("myProjectKey", fileName).initMetadata(content));
+    InputFile cxxFile = sensorContext.fileSystem().inputFile(sensorContext.fileSystem().predicates().hasPath(fileName));
+    
+    SourceFile file = CxxAstScanner.scanSingleFile(cxxFile, sensorContext, check);  
+
+    
     CheckMessagesVerifier.verify(file.getCheckMessages())
       .next().atLine(2).withMessage(check.message)
       .next().atLine(3).withMessage(check.message)
       .noMore();
   }
 
   @Test
-  public void lineRegExInvertWithoutFilePattern() {
+  public void lineRegExInvertWithoutFilePattern() throws UnsupportedEncodingException, IOException {
     LineRegularExpressionCheck check = new LineRegularExpressionCheck();
     check.regularExpression = "//.*";
     check.invertRegularExpression = true;
     check.message = "Found no comment in the line!";
 
-    SourceFile file = CxxAstScanner.scanSingleFile(new File("src/test/resources/checks/LineRegExInvert.cc"), check);
+    String fileName = "src/test/resources/checks/LineRegExInvert.cc";
+    SensorContextTester sensorContext = SensorContextTester.create(new File("."));
+    String content = new String(Files.readAllBytes(new File(sensorContext.fileSystem().baseDir(), fileName).toPath()), "UTF-8");
+    sensorContext.fileSystem().add(new DefaultInputFile("myProjectKey", fileName).initMetadata(content));
+    InputFile cxxFile = sensorContext.fileSystem().inputFile(sensorContext.fileSystem().predicates().hasPath(fileName));
+    
+    SourceFile file = CxxAstScanner.scanSingleFile(cxxFile, sensorContext, check);  
+
+    
     CheckMessagesVerifier.verify(file.getCheckMessages())
       .next().atLine(3).withMessage(check.message)
       .noMore();
   }
 
   @Test
-  public void lineRegExWithFilePattern1() {
+  public void lineRegExWithFilePattern1() throws UnsupportedEncodingException, IOException {
     LineRegularExpressionCheck check = new LineRegularExpressionCheck();
     check.matchFilePattern = "/**/*.cc"; // all files with .cc file extension
     check.regularExpression = "#include\\s+\"stdafx\\.h\"";
     check.message = "Found '#include \"stdafx.h\"' in line in a .cc file!";
 
-    SourceFile file = CxxAstScanner.scanSingleFile(new File("src/test/resources/checks/LineRegEx.cc"), check);
+    String fileName = "src/test/resources/checks/LineRegEx.cc";
+    SensorContextTester sensorContext = SensorContextTester.create(new File("."));
+    String content = new String(Files.readAllBytes(new File(sensorContext.fileSystem().baseDir(), fileName).toPath()), "UTF-8");
+    sensorContext.fileSystem().add(new DefaultInputFile("myProjectKey", fileName).initMetadata(content));
+    InputFile cxxFile = sensorContext.fileSystem().inputFile(sensorContext.fileSystem().predicates().hasPath(fileName));

```

Comment:
@jmecosta This block is very often in tests. Maybe we can encapsulate it in a helper?
```
String fileName = "src/test/resources/checks/LineRegEx.cc";
SensorContextTester sensorContext = SensorContextTester.create(new File("."));
String content = new String(Files.readAllBytes(new File(sensorContext.fileSystem().baseDir(), fileName).toPath()), "UTF-8");
sensorContext.fileSystem().add(new DefaultInputFile("myProjectKey", fileName).initMetadata(content));
InputFile cxxFile = sensorContext.fileSystem().inputFile(sensorContext.fileSystem().predicates().hasPath(fileName));
```